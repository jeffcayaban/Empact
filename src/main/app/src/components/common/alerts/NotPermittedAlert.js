import React from "react";
import {Alert, Button} from "react-bootstrap";
import {Route} from "react-router-dom";

/**
 * Is an alert for informing the user that they are not permitted to view a specific page.
 */

class NotPermittedAlert extends React.PureComponent {
    render() {
        return (
            <div className={"alertContainer"}>
                <Alert bsStyle={"danger"}>
                    <h4>You are not permitted to view this page.</h4>
                    <p>Click on the button below to explore the petitions on the website!</p>
                    <div style={{ marginTop: '15px' }}>
                        <Route render={({ history }) => (
                            <Button onClick={() => { history.push('/petitions') }}>
                                Explore
                            </Button>
                        )}/>
                    </div>
                </Alert>
            </div>
        );
    }
}

export default NotPermittedAlert;