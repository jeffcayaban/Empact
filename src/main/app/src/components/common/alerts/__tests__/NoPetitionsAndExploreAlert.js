import React from 'react';
import toJson from "enzyme-to-json";
import { shallow } from "enzyme";
import NotPermittedAlert from "../NotPermittedAlert";

describe('NotPermittedAlert', () => {

    it('should render successfully given a showFollowUp context', () => {
        const component = shallow(<NotPermittedAlert />);
        expect(toJson(component)).toMatchSnapshot();
    });
});
