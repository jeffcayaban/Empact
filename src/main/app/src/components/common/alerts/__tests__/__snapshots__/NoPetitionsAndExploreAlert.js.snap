// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotPermittedAlert should render successfully given a showFollowUp context 1`] = `
<div
  className="alertContainer"
>
  <Alert
    bsClass="alert"
    bsStyle="danger"
    closeLabel="Close alert"
  >
    <h4>
      You are not permitted to view this page.
    </h4>
    <p>
      Click on the button below to explore the petitions on the website!
    </p>
    <div
      style={
        Object {
          "marginTop": "15px",
        }
      }
    >
      <Route
        render={[Function]}
      />
    </div>
  </Alert>
</div>
`;
